homeassistant/components/lacrosse_view/sensor.py

```diff
@@ -116,21 +148,31 @@ def __init__(
     ) -> None:
         """Initialize."""
         super().__init__(coordinator)
+
         sensor = self.coordinator.data[int(description.key)]
 
+        if not sensor.permissions.get("read"):
```

Comment:
This check should be done before creating the entity.

Should we start a reauth flow in this case? How can the user resolve this?

Reply:
This API is one of those cases where this is an undocumented API discovered through proxying the app's requests.
I've never had a scenario where this error happens, but if this does happen, I'd recommend having the user just create a new config with the account that has permissions to read the sensor. I don't know if we even need this.

Reply:
It sounds like we should start a reauth flow to tell the user about this and let them authenticate again, fixing the problem.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#reauthentication

Besides `raising ConfigEntryAuthFailed` we can start a reauth flow with this method:

https://github.com/home-assistant/core/blob/6ad27089468e551ce4fb33021684c5366ce1d4f9/homeassistant/config_entries.py#L642-L649

Reply:
The reauth flow can be done in a separate PR. For now, here, just log an error and do not create any entities for that sensor with incorrect permissions.

Reply:
Do I remove the reauth flow and move it to a separate PR or raise `ConfigEntryAuthFailed`?

Reply:
Sorry, I forgot we've already added a reauth flow in this PR. Then it's ok to log a warning, start the reauth flow here and return.

Reply:
I think it would be easier to check this in the coordinator update method. Then we can just raise `ConfigEntryAuthFailed`.